src/foundry/client/applications/ux/form-data-extended.d.mts

```diff
@@ -8,10 +10,80 @@
  *  Support for defined data types and type conversion
  *  Support for TinyMCE editors
  *  Support for editable HTML elements
- * @remarks TODO: Stub, copy from v12 implementation & update
  */
-declare class FormDataExtended extends FormData {}
+declare class FormDataExtended extends FormData {
+  /**
+   * @param form    - The form being processed
+   * @param options - Options which configure form processing
+   */
+  constructor(form: HTMLFormElement, options?: FormDataExtended.Options);
+
+  /**
+   * A mapping of data types requested for each form field.
+   * @defaultValue `{}`
+   */
+  dtypes: Partial<Record<string, string>>;
+
+  /**
+   * A record of TinyMCE editors which are linked to this form.
+   * @defaultValue `{}`
+   */
+  editors: Partial<Record<string, foundry.appv1.api.FormApplication.FormApplicationEditor>>;
```

Comment:
`Partial<Record<T, U>>` isn't that useful. We recommend `noUncheckedIndexedAccess` though which makes this not useful at all.